File: problems/2507_smallest_value_after_replacing_with_sum_of_prime_factors/Solution.java
class Solution {
    public int smallestValue(int n) {
        while (true) {
            int s = 0, x = n;
            for (int d = 2; d * d <= x; d++) while (x % d == 0) { s += d; x /= d; }
            if (x > 1) s += x;
            if (s == n) return n;
            n = s;
        }
    }
}

File: problems/2507_smallest_value_after_replacing_with_sum_of_prime_factors/solution.py
class Solution:
    def smallestValue(self, n: int) -> int:
        while True:
            s, x = 0, n
            d = 2
            while d * d <= x:
                while x % d == 0:
                    s += d
                    x //= d
                d += 1
            if x > 1:
                s += x
            if s == n:
                return n
            n = s
